test/support/flutter_test_environment.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:devtools/globals.dart';
+import 'package:devtools/service_manager.dart';
+import 'package:devtools/utils.dart';
+import 'package:devtools/vm_service_wrapper.dart';
+
+import 'flutter_test_driver.dart';
+
+/// Switch this flag to false to debug issues with non-atomic test behavior.
+bool reuseTestEnvironment = true;
+
+class FlutterTestEnvironment {
+  FlutterTestEnvironment(this._runConfig);
+
+  FlutterRunConfiguration _runConfig;
+  FlutterRunTestDriver _flutter;
+  FlutterRunTestDriver get flutter => _flutter;
+  VmServiceWrapper _service;
+  VmServiceWrapper get service => _service;
+
+  // This function will be called before creating and running the Flutter app.
+  VoidAsyncFunction _beforeSetup;
+  set beforeSetup(VoidAsyncFunction f) => _beforeSetup = f;
+
+  // This function will be called after we have ran the Flutter app and the
+  // vmService is opened.
+  VoidAsyncFunction _afterSetup;
+  set afterSetup(VoidAsyncFunction f) => _afterSetup = f;
+
+  // The function will be called before tearing down the test and stopping the
+  // Flutter app.
+  VoidAsyncFunction _beforeTearDown;
+  set beforeTearDown(VoidAsyncFunction f) => _beforeTearDown = f;
+
+  bool _needsSetup = true;
+
+  Future<void> setupEnvironment({bool force = false}) async {
+    if (!force && !_needsSetup && reuseTestEnvironment) {
+      // Setting up the environment is slow so we reuse the existing environment
+      // when possible.
+      return;
+    }
+    if (_beforeSetup != null) await _beforeSetup();
+
+    _flutter = FlutterRunTestDriver(Directory('test/fixtures/flutter_app'));
```

Comment:
bonus points: make this the default value of a named parameter of this class.

Reply:
Done.